### PR TITLE
handlers/ directory, and reorg

### PR DESCRIPTION
**Pull request by sbenthall**
addressing #16

### CODE REVIEW

**Comment by sbenthall** [comment]
@benheller I'm wondering how seriously to take this cross-site scripting vulnerability identified by CodeQL, and if you see any easy workarounds (e.g. sanitization). In my view, CodeQL is being a little too aggressive here, but maybe I'm missing something.

**Comment by sbenthall** [comment]
Actually, in the interest of tight PRs, probably best to review/merge at this stage.

This PR:
 - reorganizes the `discord/` directory as per #16 
 - creates a `handlers/` directory the same level
 - populated `handlers/` with a handler dedicated to drawing a card in the game.

This PR does not complete #16 by changing the discord endpoints to use handlers shared with the HTTP routes. That can come in a later PR.

File: app/handlers/draw.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    const cards = await Card
+      .aggregate()
+      .match({ _id: { $nin: blacklist }, deck: deck._id })
+      .sample(1);
+  
+    const card = cards[0];
+  
+    await Game.updateOne({ _id: game._id }, { $push: { cards: card._id }});
+
+    log.debug(card)
```
Comment (benheller) [inline app/handlers/draw.js:25]: Some minor linting issues here:
1. missing semicolon on this line and the line below
1. missing newline at end of file

File: app/routes/game.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import Card from '../models/Card.js';
-import Game from '../models/Game.js';
 import Deck from '../models/Deck.js';
 
+import gameDraw from '../handlers/draw.js'
```
Comment (benheller) [inline app/routes/game.js:3]: nit: missing semicolon, run `eslint` to find all these

File: app/routes/game.js [hunk 3]
```diff
@@ -41,27 +41,12 @@ export async function draw(req, res) {
     return res.send(`Deck ${deckName} does not exist!`);
```
Comment (benheller) [inline app/routes/game.js:41]: Yep, that's a legit warning. I wrote it, so shame on me. Go ahead and remove the `deckName` variable from this return statement, or spend some time investigating what kind of sanitization is necessary here.

File: app/routes/game.js [hunk 4]
```diff
@@ -41,27 +41,12 @@ export async function draw(req, res) {
...
-
-  const blacklist = game.cards;
-
-  const cards = await Card
-    .aggregate()
-    .match({ _id: { $nin: blacklist }, deck: deck._id })
-    .sample(1);
-
-  const card = cards[0];
+  const card = await gameDraw(deck)
```
Comment (benheller) [inline app/routes/game.js:44]: nit: missing semicolon

**Review by benheller — APPROVED** [review]
Generally looks good. Run `eslint` and fix the cross-site-scripting issue and it's good to go.

**Comment by sbenthall** [comment]
I don't know how to get eslint working locally, see #38 
Happy to do it if there's working instructions!

This cross-site scripting vulnerability looks like it was introduced in the initial commit and is not related to the changes in this PR. Could you resolve it?
https://github.com/yrevocnu/psychedelic-glass-bead-game/blame/master/app/routes/game.js#L29-L43

I will address your nits and fixup this PR.

File: app/routes/game.js (hunk 3 above)
Comment (benheller) [reply app/routes/game.js:41]: @sbenthall This has now been addressed on `master`

**Review by benheller — COMMENTED** [review]
Generally speaking, is there a reason you're working from a `fork` rather than creating feature branches within this repository?

**Comment by sbenthall** [comment]
@benheller I've ran the linter. This now requires your review to merge.

Please note I had to `git commit -n` this to get past the linter, which was complaining based on some issues unrelated with this feature. See #42 

Fork and pull is a fairly common development pattern for open source projects.

If you asked me to justify it as an alternative to feature branches, I could come up with a few reasons:
 - Prevents cluttering of feature branches in the core repository
 - Means any messy git history and/or broken rebases need review before making it into core repository

**Comment by sbenthall** [comment]
For me, merging this PR is blocked by your review, unless I override the project settings with admin privileges.

image

I'm not sure what your intention is here. Perhaps you could approve this PR?

**Review by benheller — APPROVED** [review]
Approved! I added some basic restrictions to the repo (which you, as an administrator, can ignore) to enforce best practices when we expand our base of contributors.

**Comment by sbenthall** [comment]
I cannot merge this PR because the commit is not signed.
This is a new requirement.
I will need to figure out how to sign commits and get back to this later.

**Comment by benheller** [comment]
I have disabled the requirement to sign commits. That said, please proceed in figuring out how to do this, as it would be good to re-enable it so that contributors to this project cannot be impersonated. Let me know when you're all set there and I'll flip it back on 👍

**Comment by sbenthall** [comment]
Ok, thanks. I'll merge this one.
I'll look into configuring Git/Hub to sign my commits and let you know when that is set up.

In the meantime, if you have any doubts about whether I've authored a commit here, I welcome you to use a 2FA method. You can text me, and I can text you back a personal detail that only I would know about you, along with the commit hash.

**Pull request merged**